File: gettls/week13/[3986] 좋은 단어.cpp
#include <string>
#include <map>
#include <stack>
#include <queue>
#include <vector>
#include <algorithm>
#include <cstring>
#include <iostream>
#include <sstream>
using namespace std;
#pragma warning (disable:4996)
int main() {
	int n;
	int cnt = 0;
	cin >> n;
	for (int i = 0; i < n; i++) {
		string str;
		stack<char> s;
		cin >> str;
		for (int i = 0; i < str.size(); i++) {
			if (s.empty()) {
				s.push(str[i]); continue;
			}

			if (s.top() == str[i]) s.pop();
			else s.push(str[i]);
		}

		if (s.empty()) cnt++;

	}
	cout << cnt;
}

File: wonseok970/문자열/[3986] 좋은 단어.py
N = int(input())
lst = [input() for _ in range(N)]
cnt = 0

for word in lst:
    # 홀수의 경우
    if len(word) % 2 != 0:
        continue
    else:
        stack = [word[0]]

    for i in range(1, len(word)):
        if len(stack) == 0:
            stack.append(word[i])
            continue
        if stack[-1] == word[i]:
            stack.pop()
        else:
            stack.append(word[i])
    # 짝을 모두 찾았을 떄,
    if len(stack) == 0:
        cnt += 1
    else:
        continue

print(cnt)
